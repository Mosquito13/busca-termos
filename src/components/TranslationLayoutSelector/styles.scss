@import '../../sass/styles.scss';

.translation-layout-selector {
  &__custom-content {
    background-color: transparent;
    flex: 1;
    height: 100%;
    display: flex;
    border: 2px solid currentColor;
    border-radius: $radius;

    & > span {
      flex: 1;
    }

    & > span:not(:last-child) {
      border-right: 2px solid currentColor;
    }
  }
}
