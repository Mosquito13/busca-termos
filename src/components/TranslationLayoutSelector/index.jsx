import PropTypes from 'prop-types';
import times from 'lodash/times';

import ButtonRadio from '../common/ButtonRadio';

import './styles.scss';

const getCustomContent = value => (
  <div className="translation-layout-selector__custom-content">
    {times(value, (v) => <span key={`span_${v}`} />)}
  </div>
);

const TranslationLayoutSelector = ({ value, onChange }) => {
  return (
    <div className="translation-layout-selector">
      <ButtonRadio
        value={value}
        onChange={onChange}
        options={[
          {
            value: 1,
            text: 'Uma coluna',
            customContent: getCustomContent(1)
          },
          {
            value: 2,
            text: 'Duas colunas',
            customContent: getCustomContent(2)
          },
          {
            value: 3,
            text: 'TrÃªs colunas',
            customContent: getCustomContent(3)
          }
        ]}
      />
    </div>
  );
};

TranslationLayoutSelector.propTypes = {
  /**
   * Valor
   */
  value: PropTypes.number,
  /**
   * Funcao para alterar o valor
   */
  onChange: PropTypes.func
};

export default TranslationLayoutSelector;
