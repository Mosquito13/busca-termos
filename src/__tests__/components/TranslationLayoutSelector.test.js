import renderer from 'react-test-renderer';

import TranslationLayoutSelector from '../../components/TranslationLayoutSelector';

describe(`${TranslationLayoutSelector}`, () => {
  describe('snapshots', () => {
    it('should render', () => {
      const tree = renderer
        .create(<TranslationLayoutSelector value={2} onChange={jest.fn()} />)
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
