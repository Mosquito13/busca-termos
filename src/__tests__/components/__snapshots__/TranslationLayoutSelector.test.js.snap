// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TranslationLayoutSelector snapshots should render 1`] = `
<div
  className="translation-layout-selector"
>
  <div
    className="button-radio"
  >
    <button
      className="button-radio__option"
      onClick={[Function]}
      title="Uma coluna"
    >
      <div
        className="translation-layout-selector__custom-content"
      >
        <span />
      </div>
    </button>
    <button
      className="button-radio__option button-radio__option--selected"
      onClick={[Function]}
      title="Duas colunas"
    >
      <div
        className="translation-layout-selector__custom-content"
      >
        <span />
        <span />
      </div>
    </button>
    <button
      className="button-radio__option"
      onClick={[Function]}
      title="TrÃªs colunas"
    >
      <div
        className="translation-layout-selector__custom-content"
      >
        <span />
        <span />
        <span />
      </div>
    </button>
  </div>
</div>
`;
